web/src/OverviewActionBar.tsx

```diff
@@ -326,6 +411,94 @@ export function FilterRadioButton(props: FilterRadioButtonProps) {
   )
 }
 
+export const FILTER_INPUT_DEBOUNCE = 500 // in ms
+export const FILTER_FIELD_ID = "FilterTermTextInput"
+
+const debounceFilterLogs = debounce((history: History, search: string) => {
+  history.push({ pathname: location.pathname, search })
+}, FILTER_INPUT_DEBOUNCE)
+
+export function FilterTermField(props: { filterSet: FilterSet }) {
+  // The global term is set from the url, which is passed
+  // down through props and used to set the initial value
+  const globalTerm = props.filterSet.term
+  const [filterTerm, setFilterTerm] = useState(globalTerm.input ?? EMPTY_TERM)
+
+  const history = useHistory()
+  const location = history.location
+
+  const inputProps: InputProps = {
+    startAdornment: (
+      <InputAdornment position="start" disablePointerEvents={true}>
+        <FilterSvg fill={Color.grayDark} role="presentation" />
+      </InputAdornment>
+    ),
+  }
+
+  // If there's a search term, add a button to clear that term
+  if (filterTerm) {
+    const onClearClick = () => {
+      // Clear the input field's value
```

Comment:
I'd be tempted to replace this and the onChange function to a call to something like this:
```
  const setTerm = (term: string) => {
    setFilterTerm(term)
    const search = createLogSearch(location.search, { term })
    if (term === EMPTY_TERM) {
      history.push({
        pathname: location.pathname,
        search: search.toString(),
      })
    } else {
      /**
       * Note: debouncing allows us to wait to execute log filtration until a set
       * amount of time has passed without the filter term changing. To implement
       * debouncing, it's necessary to separate the term field's value from the url
       * search params, otherwise the field that a user types in doesn't update.
       * The term field updates without any debouncing, while the url search params
       * (which actually triggers log filtering) updates with the debounce delay.
       */
      debounceFilterLogs(history, search.toString())
    }
  }
```
just since there are a number of things going on and I found myself wondering if they both did all the same things

Reply:
That's fair! I've combined the logic like you've outlined and removed the debouncing delay when there's an empty term that will be the new state.